dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/LuceneQueryDateTimeFormatter.java

```diff
@@ -0,0 +1,255 @@
+package com.dotcms.content.elasticsearch.business;
+
+import static com.dotcms.content.elasticsearch.business.ESMappingAPIImpl.dateFormat;
+import static com.dotcms.content.elasticsearch.business.ESMappingAPIImpl.datetimeFormat;
+
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.cache.FieldsCache;
+import com.dotmarketing.portlets.structure.model.Field;
+import com.dotmarketing.portlets.structure.model.Structure;
+import com.dotmarketing.util.DateUtil;
+import com.dotmarketing.util.RegEX;
+import com.dotmarketing.util.RegExMatch;
+import com.dotmarketing.util.UtilMethods;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ *
+ * @author nollymar
+ */
+public class LuceneQueryDateTimeFormatter {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A class which only has private constructors should be final](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4545506)